From ee4a64355de04fc8dcf0de328d3a22f378900114 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hmehrtens@maxlinear.com>
Date: Thu, 17 Jun 2021 18:45:34 +0200
Subject: [PATCH] PONRTSYS-9036: Make use of tcf_vlan_push_flags()

Use the function tcf_vlan_push_flags() on PRX300 and LGM.
---
 drivers/net/datapath/pon_qos/pon_qos_tc_vlan_prepare.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_vlan_prepare.c b/drivers/net/datapath/pon_qos/pon_qos_tc_vlan_prepare.c
index dc177e356641..299609adbf38 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_vlan_prepare.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_vlan_prepare.c
@@ -294,16 +294,9 @@ static void vlan_action_vid_parse(struct net_device *dev,
 				  enum tc_flower_vlan_tag tag,
 				  int *vid)
 {
-	struct tcf_vlan *v = to_vlan(a);
-#if (KERNEL_VERSION(4, 15, 0) <= LINUX_VERSION_CODE)
-	struct tcf_vlan_params *p;
+	u8 flags = tcf_vlan_push_flags(a);
 
-	p = rcu_dereference_bh(v->vlan_p);
-
-	if (p->tcfv_push_flags & ACTVLAN_PUSH_F_ID) {
-#else
-	if (v->tcfv_push_flags & ACTVLAN_PUSH_F_ID) {
-#endif
+	if (flags & ACTVLAN_PUSH_F_ID) {
 		*vid = tcf_vlan_push_vid(a);
 		return;
 	}
